### ISSUE TITLE
"[C]apture variables are only available outside the regex" is an outdated info?

### ISSUE THREAD

**Issue by zacque0** [Raku/doc#4113]
## Problem or new feature

The quote in the title can be found here: https://github.com/Raku/doc/blob/c9cbe0fa588883febb073d4677b613f451dcd4db/doc/Language/regexes.pod6#L1507-L1529

On my machine, both
```raku
say "11" ~~ / (\d) $0 /; # OUTPUT: «｢11｣␤ 0 => ｢1｣␤ 1 => ｢1｣␤»
say "11" ~~ / (\d) {} $0 /; # OUTPUT: «｢11｣␤ 0 => ｢1｣␤ 1 => ｢1｣␤»
```
returns the same result. Here's a more complicated example:
```raku
say "12121212" ~~ / ((\d) (\d) $0 $1) $0 /; # Output: «｢12121212｣␤0 => ｢1212｣␤0 => ｢1｣␤1 => ｢2｣␤»
```

Environment:
x86_64, Linux.

`raku -v`:
```
Welcome to Rakudo™ v2022.07.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2022.07.
```

## Suggestions

State whether the doc outdated or this is an undocumented feature in Rakudo. Then, perhaps the link section above can be rephrased because it seems complicated and trying to address a different problem. Thanks!

**Comment by raiph** [comment]
> These capture variables are only available outside the regex.

Yeah, that sentence seems just plain wrong. I don't think it was ever true.

I think it's an attempt to document the fact that "sub-captures" (child captures) are stored ***inside** their innermost enclosing (parent) **capture***.

It's not *outside*, and it's not outside *a regex*.

```
# !!WRONG!! The $0 refers to a capture *inside* the second capture 
say "11" ~~ /(\d) ($0)/; # OUTPUT: «Nil␤» 
```
The code comment is rather sloppy but at least it's right. If you already know what it's trying to convey, you can tell what it's trying to convey. If you don't, it's likely really confusing.

> In order to make them available inside the Regexes, you need to insert a code block behind the match; this code block may be empty if there's nothing meaningful to do

That's a different issue. One is to do with nesting. This one is to do with "Publication of match variables".

I think the thing to do to know what you need to know to fix this doc is to read and understand enough of the answers to the SO [Why/how is an additional variable needed in matching repeated arbitary character with capture groups?](https://stackoverflow.com/questions/56393888/why-how-is-an-additional-variable-needed-in-matching-repeated-arbitary-character). I think the best thing to do would be to read all of them until you understand enough to fix this doc.

This SO deals with both the two issues covered by the doc you've started with. Perhaps that's why they were written as they are.
